nuxeo-runtime/nuxeo-runtime-aws/src/main/java/org/nuxeo/runtime/aws/NuxeoAWSCredentialsProviderImpl.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (C) Copyright 2015-2018 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Remi Cattiau
+ */
+package org.nuxeo.runtime.aws;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+
+public class NuxeoAWSCredentialsProviderImpl extends DefaultComponent implements NuxeoAWSCredentialsProvider {
+
+    private AWSCredentialsDescriptor credentials = null;
+
+    public boolean hasContributionCredentials() {
+        return credentials != null;
+    }
+
+    public AWSCredentialsProvider getProvider() {
+        if (hasContributionCredentials()) {
+            return new AWSStaticCredentialsProvider(new BasicAWSCredentials(credentials.getAWSAccessKeyId(), credentials.getAWSSecretKey()));
+        }
+        return new DefaultAWSCredentialsProviderChain();
+    }
+
+    @Override
+    public void registerContribution(Object contribution,
+                                     String extensionPoint, ComponentInstance contributor) {
+        if ("credentials".equals(extensionPoint)) {
+            credentials = (AWSCredentialsDescriptor) contribution;
+        }
+    }
+
```

Comment:
Can you please comply to new way of writing a Component ?
You can see there an example there : https://github.com/nuxeo/nuxeo/blob/fa7eb83350647d0bcc95727704e837bfb9a0a0bd/nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroComponent.java